shipyard_blueprints/microsoft-onedrive/shipyard_microsoft_onedrive/utils.py

```diff
@@ -0,0 +1,29 @@
+from shipyard_templates import ShipyardLogger, ExitCodeException, CloudStorage
+
+logger = ShipyardLogger.get_logger()
+
+
+def get_credential_group(args):
+    if args.access_token:
+        logger.debug("Using access token for authentication")
+        return {"access_token": args.access_token}
+    if args.client_id and args.client_secret and args.tenant:
+        logger.debug("Using Client ID, Client Secret, and Tenant for authentication")
+        return {
+            "client_id": args.client_id,
+            "client_secret": args.client_secret,
+            "tenant": args.tenant,
+        }
+    # if args.client_id and args.tenant and args.username and args.password:
```

Comment:
Currently not supported in our blueprints but added here for an easy patch when we roll out the credential group support